Compilation error related to symmetric ICP objective

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Operating System and version: Ubuntu 18.04
* Compiler: GCC 7.4
* PCL Version: master

Got the following error compiling a downstream project with the fresh master:

```cpp
/usr/local/include/pcl-1.9/pcl/registration/icp.h:347:13: error: no matching function for call to ‘boost::shared_ptr<pcl::registration::TransformationEstimation<pcl::PointXYZRGBNormal, pcl::PointXYZRGBNormal, double> >::reset(boost::shared_ptr<pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<pcl::PointXYZRGBNormal, pcl::PointXYZRGBNormal, double> >&)’
             transformation_estimation_.reset (symmetric_transformation_estimation);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is clearly caused by #3390. The problem and the fix are obvious, but what needs to be checked is how this escaped surfacing during unit tests.

--- a/registration/include/pcl/registration/icp.h
+++ b/registration/include/pcl/registration/icp.h
@@ -41,6 +41,7 @@
 #pragma once
 
 // PCL includes
+#include <pcl/make_shared.h>
 #include <pcl/sample_consensus/ransac.h>
 #include <pcl/sample_consensus/sac_model_registration.h>
 #include <pcl/registration/registration.h>
@@ -342,9 +343,9 @@ namespace pcl
         use_symmetric_objective_ = use_symmetric_objective;
         if (use_symmetric_objective_)
         {
-            auto symmetric_transformation_estimation = std::make_shared<pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar> > ();
+            auto symmetric_transformation_estimation = pcl::make_shared<pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar> > ();
             symmetric_transformation_estimation->setEnforceSameDirectionNormals (enforce_same_direction_normals_);
-            transformation_estimation_.reset (symmetric_transformation_estimation);
+            transformation_estimation_ = symmetric_transformation_estimation;
         }
         else
         {

--- a/registration/include/pcl/registration/impl/transformation_estimation_symmetric_point_to_plane_lls.hpp
+++ b/registration/include/pcl/registration/impl/transformation_estimation_symmetric_point_to_plane_lls.hpp
@@ -151,8 +151,8 @@ estimateRigidTransformation (ConstCloudIterator<PointSource>& source_it, ConstCl
   {
     const Vector3 p (source_it->x, source_it->y, source_it->z);
     const Vector3 q (target_it->x, target_it->y, target_it->z);
-    const Vector3 n1 (source_it->getNormalVector3fMap());
-    const Vector3 n2 (target_it->getNormalVector3fMap());
+    const Vector3 n1 (source_it->getNormalVector3fMap().template cast<Scalar>());
+    const Vector3 n2 (target_it->getNormalVector3fMap().template cast<Scalar>());
     Vector3 n;
     if (enforce_same_direction_normals_)
     {

--- a/test/registration/test_registration.cpp
+++ b/test/registration/test_registration.cpp
@@ -191,6 +191,17 @@ TEST (PCL, IterativeClosestPoint)
 //  EXPECT_EQ (transformation (3, 3), 1);
 }
 
+TEST (PCL, IterativeClosestPointWithNormals)
+{
+  IterativeClosestPointWithNormals<PointNormal, PointNormal, float> reg_float;
+  reg_float.setUseSymmetricObjective(true);
+  EXPECT_TRUE(reg_float.getUseSymmetricObjective());
+
+  IterativeClosestPointWithNormals<PointNormal, PointNormal, double> reg_double;
+  reg_double.setUseSymmetricObjective(true);
+  EXPECT_TRUE(reg_double.getUseSymmetricObjective());
+}
+
 ////////////////////////////////////////////////////////////////////////////////////////////////////////
 void
 sampleRandomTransform (Eigen::Affine3f &trans, float max_angle, float max_trans)

